boards/arm/mimxrt1010_evk/doc/index.rst

```diff
@@ -191,6 +191,12 @@ see the following message in the terminal:
 
     Hello World! mimxrt1010_evk
 
+MCUXpresso Config Tool
+======================
+
+A ``mimxrt1010_evk.mex`` file is included. This file was used to generate the clock
+initialization code and can be used as a starting point to tweak the clock configuration.
+This could be useful for different boards that are based on i.MX RT1010
```

Comment:
I would rather expect a "Clock start up configuration" chapter which explains what is the clock configuration provided (all clocks off, if not which ones are active, ...).
In that chapter, you can explain how this was generated, why it is fine vs Zephyr policy and next steps (conversion to device tree).

Reply:
> vs Zephyr policy and next steps (conversion to device tree).

I think I agree: if it is explicitly documented that this "should" be in devicetree and that you're working towards that, I won't block either.

@dleach02 's can you confirm that the current code in the tree is:

1. wrong,
2. needs to be made board-specific in some way

and

3. NXP's goal is to convert it to a common set of devicetree-based driver(s) that can get the board-specific clock configuration from the board .dts files?

I think that's what you were saying on Discord; I just want to confirm.

Reply:
@mmahadevan108, lets update the documentation and also move the .mex file to the HAL (or document how a user can get the file from the NXP sdk).

Reply:
@mbolivar-nordic, 

1. Yes. We think when the boards were initially added to the Zephyr project that it copied the wrong values from the SDK. Further, once we started plugging in the values into our configuration tool we found additional error in a clock value that was out of spec.
2. the configuration values for the clock stuff needs to be at the board level. Consider that the crystal frequency selection is a board specific property which impacts the values being used. This will be replaced with a devicetree in the board folder with the specific values needed. 
3. Our goal is to transition all of our in-tree supported boards to device tree.

Reply:
@mbolivar-nordic 

(Trying this reply again)

1. Yes, the original code is wrong. We think that maybe the original code copied over from the SDK was incorrect. We further noted that our configuration tool highlighted an incorrect value that was out of specification.
2. Needs to be board specific. Consider that the clock frequency selection is a board specific property that impacts the values being used by the clock configuration. This will eventually be replaced by a board specific devicetree with these values.
3. Yes, we will transition dts

Reply:
Also, what would be nice is to add a reference to a tracking issue.

Reply:
I have updated the documentation to include a table with the clock configuration on platform initialization. Also a note has been added regarding future device tree support.